### ISSUE TITLE
How to train on Sun80  or Urban100?

### ISSUE THREAD

**Issue by Lkinyuu** [dvlab-research/MASA-SR#13]
This is my first time learning RefSR. Sun80 or Urban100 sounds like SISR dataset. May I  know how to train RefSR code on them? Thanks!

**Comment by SkyeLu** [comment]
Hi, as mentioned in Sec. 4.1 in our paper, our model is only trained on CUFED5 training set, and then tested on other datasets. Thus Sun80 or Urban100 are only used for testing not training.

**Comment by Lkinyuu** [comment]
> Hi, as mentioned in Sec. 4.1 in our paper, our model is only trained on CUFED5 training set, and then tested on other datasets. Thus Sun80 or Urban100 are only used for testing not training.

Thanks ！ Another question is how to train or test the SISR model？ I saw that the SISR models in the paper were also trained on the CUFED5 .Can I know the relevant steps or parameters?

**Comment by Lkinyuu** [comment]
> Hi, as mentioned in Sec. 4.1 in our paper, our model is only trained on CUFED5 training set, and then tested on other datasets. Thus Sun80 or Urban100 are only used for testing not training.

Hello, there are both HR and LR png in Urban100 and Sun80 datasets. So maybe you can introduce the settings in dataset.py？                                                                                                                                                                   
for example: self.input_list = sorted(glob.glob('/home/liyinglu/newData/datasets/SR/SR_testing_datasets/Sun_Hays_SR_groundtruth/*.jpg'))
        self.ref_folders = '/home/liyinglu/newData/datasets/SR/SR_testing_datasets/Sun_Hays_SR_scenematches'

**Comment by Lkinyuu** [comment]
@SkyeLu

**Comment by Lkinyuu** [comment]
In TTSR issue,for Urban100 and Manga109 which lacks of ref, ref is just the lr itself, which means ref_sr = lr↓↑. But in the dataset of MASA-SR,it looks like just input hr ，LRand ref are generated from HR

**Comment by ximu1211** [comment]
> Hi, as mentioned in Sec. 4.1 in our paper, our model is only trained on CUFED5 training set, and then tested on other datasets. Thus Sun80 or Urban100 are only used for testing not training.
output_img = calculate_PSNR_SSIM.bgr2ycbcr(output_img, only_y=to_y)
gt = calculate_PSNR_SSIM.bgr2ycbcr(gt, only_y=to_y)
When testing on Sun80 or Urban100 ,should I delete the code ?                                     As far as I understand, only CUFED5 needs   YCbCr color space.

**Comment by ximu1211** [comment]
> Hi, as mentioned in Sec. 4.1 in our paper, our model is only trained on CUFED5 training set, and then tested on other datasets. Thus Sun80 or Urban100 are only used for testing not training.

output_img = calculate_PSNR_SSIM.bgr2ycbcr(output_img, only_y=to_y)
gt = calculate_PSNR_SSIM.bgr2ycbcr(gt, only_y=to_y)
When testing on Sun80 or Urban100 ,should I delete the code ?                                     As far as I understand, only CUFED5 needs   YCbCr color space.

**Comment by SkyeLu** [comment]
> > Hi, as mentioned in Sec. 4.1 in our paper, our model is only trained on CUFED5 training set, and then tested on other datasets. Thus Sun80 or Urban100 are only used for testing not training.
> 
> Thanks ！ Another question is how to train or test the SISR model？ I saw that the SISR models in the paper were also trained on the CUFED5 .Can I know the relevant steps or parameters?

Hi, sorry for the late reply, the SISR model architectures can be found in their offical implementations. You can add those architectures  into our `models/archs` folder and make some slight modifications to the training code. The training settings are same as those for training MASA-rec, except that the reference image is not used.

**Comment by SkyeLu** [comment]
> > Hi, as mentioned in Sec. 4.1 in our paper, our model is only trained on CUFED5 training set, and then tested on other datasets. Thus Sun80 or Urban100 are only used for testing not training.
> 
> output_img = calculate_PSNR_SSIM.bgr2ycbcr(output_img, only_y=to_y) gt = calculate_PSNR_SSIM.bgr2ycbcr(gt, only_y=to_y) When testing on Sun80 or Urban100 ,should I delete the code ? As far as I understand, only CUFED5 needs YCbCr color space.

No need. All the results of PSNR and SSIM are evaluated on the Y channel of YCbCr color space.

**Comment by Lkinyuu** [comment]
> > > Hi, as mentioned in Sec. 4.1 in our paper, our model is only trained on CUFED5 training set, and then tested on other datasets. Thus Sun80 or Urban100 are only used for testing not training.
> > 
> > 
> > Thanks ！ Another question is how to train or test the SISR model？ I saw that the SISR models in the paper were also trained on the CUFED5 .Can I know the relevant steps or parameters?
> 
> Hi, sorry for the late reply, the SISR model architectures can be found in their offical implementations. You can add those architectures into our `models/archs` folder and make some slight modifications to the training code. The training settings are same as those for training MASA-rec, except that the reference image is not used.

How about the setting of loss?

**Comment by SkyeLu** [comment]
> > > > Hi, as mentioned in Sec. 4.1 in our paper, our model is only trained on CUFED5 training set, and then tested on other datasets. Thus Sun80 or Urban100 are only used for testing not training.
> > > 
> > > 
> > > Thanks ！ Another question is how to train or test the SISR model？ I saw that the SISR models in the paper were also trained on the CUFED5 .Can I know the relevant steps or parameters?
> > 
> > 
> > Hi, sorry for the late reply, the SISR model architectures can be found in their offical implementations. You can add those architectures into our `models/archs` folder and make some slight modifications to the training code. The training settings are same as those for training MASA-rec, except that the reference image is not used.
> 
> How about the setting of loss? For example, when I try to train HAN ,I copy han_arch.py and common.py ,and set only lr as input . But I failed, the loss reaches an exaggerated number

Only L1 loss is used.

**Comment by Lkinyuu** [comment]
> How about the setting of loss? For example, when I try to train HAN ,I copy han_arch.py and common.py ,and set only lr as input . But I failed, the loss reaches an exaggerated number

Thanks,is there any problem with my approach?Copy the model_arch.py ,modify the input of net. I don't know why there is such a problem

**Comment by SkyeLu** [comment]
> > How about the setting of loss? For example, when I try to train HAN ,I copy han_arch.py and common.py ,and set only lr as input . But I failed, the loss reaches an exaggerated number
> 
> Thanks,is there any problem with my approach?Copy the model_arch.py ,modify the input of net. I don't know why there is such a problem

Nothing wrong. It's weird that you got exaggerated loss value since L1 loss is a relatively stable loss. Maybe check your input and network architecture again? And also check the intermediate visualization results in the `vis` folder.

**Comment by Lkinyuu** [comment]
> > > How about the setting of loss? For example, when I try to train HAN ,I copy han_arch.py and common.py ,and set only lr as input . But I failed, the loss reaches an exaggerated number
> > 
> > 
> > Thanks,is there any problem with my approach?Copy the model_arch.py ,modify the input of net. I don't know why there is such a problem
> 
> Nothing wrong. It's weird that you got exaggerated loss value since L1 loss is a relatively stable loss. Maybe check your input and network architecture again? And also check the intermediate visualization results in the `vis` folder.

I check the intermediate visualization results in the vis folder. Strangely，the results is blank . I didn't modify any code for HAN/RCAN 's model

**Comment by Lkinyuu** [comment]
Sorry to trouble you again！@SkyeLu
When I use rbg mean in HAN(RCAN),and setting the rgb_range 255 , the results in vis folder : 
![1638776851(1)](https://user-images.githubusercontent.com/74664240/144807652-bd99bdf0-05e0-47af-8c51-b94db1b711b3.png)
When I remove the code about dataset mean in HAN(RCAN),the results in vis folder: 
![vis_0_0_0](https://user-images.githubusercontent.com/74664240/144807992-5c892737-883d-43a4-876d-7200b70c3b4d.jpg)
I didn't modify any other code .
I want to see the effect of CUFED5 on SISR, but I have encountered difficulties and cannot solve it. I will be very grateful to you if you can help me out.

**Comment by ximu1211** [comment]
I don’t know if you have made changes on dataset. Because CUFED5 works well on MASA/TTSR. But when I put it as input of code of EDSR architecture,seems something went wrong.

**Comment by SkyeLu** [comment]
> Sorry to trouble you again！@SkyeLu When I use rbg mean in HAN(RCAN),and setting the rgb_range 255 , the results in vis folder : ![1638776851(1)](https://user-images.githubusercontent.com/74664240/144807652-bd99bdf0-05e0-47af-8c51-b94db1b711b3.png) When I remove the code about dataset mean in HAN(RCAN),the results in vis folder: ![vis_0_0_0](https://user-images.githubusercontent.com/74664240/144807992-5c892737-883d-43a4-876d-7200b70c3b4d.jpg) I didn't modify any other code . I want to see the effect of CUFED5 on SISR, but I have encountered difficulties and cannot solve it. I will be very grateful to you if you can help me out.

That's weird. Did you use tensorboard to visualize the loss log in the `tb_logger/`? What does the loss look like? I just add `models/archs/RCAN_arch.py` into our code, maybe you can try to train the model in this file.

**Comment by ximu1211** [comment]
> > Sorry to trouble you again！@SkyeLu When I use rbg mean in HAN(RCAN),and setting the rgb_range 255 , the results in vis folder : ![1638776851(1)](https://user-images.githubusercontent.com/74664240/144807652-bd99bdf0-05e0-47af-8c51-b94db1b711b3.png) When I remove the code about dataset mean in HAN(RCAN),the results in vis folder: ![vis_0_0_0](https://user-images.githubusercontent.com/74664240/144807992-5c892737-883d-43a4-876d-7200b70c3b4d.jpg) I didn't modify any other code . I want to see the effect of CUFED5 on SISR, but I have encountered difficulties and cannot solve it. I will be very grateful to you if you can help me out.
> 
> That's weird. Did you use tensorboard to visualize the loss log in the `tb_logger/`? What does the loss look like? I just add `models/archs/RCAN_arch.py` into our code, maybe you can try to train the model in this file.



> > Sorry to trouble you again！@SkyeLu When I use rbg mean in HAN(RCAN),and setting the rgb_range 255 , the results in vis folder : ![1638776851(1)](https://user-images.githubusercontent.com/74664240/144807652-bd99bdf0-05e0-47af-8c51-b94db1b711b3.png) When I remove the code about dataset mean in HAN(RCAN),the results in vis folder: ![vis_0_0_0](https://user-images.githubusercontent.com/74664240/144807992-5c892737-883d-43a4-876d-7200b70c3b4d.jpg) I didn't modify any other code . I want to see the effect of CUFED5 on SISR, but I have encountered difficulties and cannot solve it. I will be very grateful to you if you can help me out.
> 
> That's weird. Did you use tensorboard to visualize the loss log in the `tb_logger/`? What does the loss look like? I just add `models/archs/RCAN_arch.py` into our code, maybe you can try to train the model in this file.

This is really weird.I found you updated repo for this stupid question of mine . And I downloaded the latest version and trained RCAN . Still cause such a situation.The loss looks normal.
![vis_0_0_0](https://user-images.githubusercontent.com/43237721/144856223-ab9bddc1-dcc3-4a6b-9cd2-7ccaeeb4196c.jpg)

**Comment by SkyeLu** [comment]
Hi, how many epochs have you trained? I guess this is a result of vis_0_*.png? @ximu1211

**Comment by ximu1211** [comment]
> Hi, how many epochs have you trained? I guess this is a result of vis_0_*.png? @ximu1211

Yeah，I think this is an embarrassing situation.I found that it was normal in the follow-up training. This is my first contact with residual based SISR model. I thought the result of the first epoch would be close to LR_SR no matter what.  I'm sorry to waste your time.

**Comment by Lkinyuu** [comment]
> > Hi, how many epochs have you trained? I guess this is a result of vis_0_*.png? @ximu1211
> 
> Yeah，I think this is an embarrassing situation.I found that it was normal in the follow-up training. This is my first contact with residual based SISR model. I thought the result of the first epoch would be close to LR_SR no matter what. I'm sorry to waste your time.

Actually I encountered the same dilemma.
